### PR TITLE
upgrade raft to v1.3.3

### PR DESCRIPTION
**Pull request by dhiaayachi**
This PR is to upgrade raft to v1.3.3 which will include https://github.com/hashicorp/raft/pull/483

### CODE REVIEW

File: .changelog/11958.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+Fix a bug where a read replica node can trigger a raft election and become a leader..
```
Comment (FFMMM) [inline .changelog/11958.txt:2]: [nit]
```suggestion
```release-note:bug
Upgrade to raft `1.3.3` which fixes a bug where a read replica node can trigger a raft election and become a leader.
```

**Review by markan — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/541980.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit d0274d38a94980d42f3ea6e5bffbdccb6903c813 onto `release/1.11.x` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit d0274d38a94980d42f3ea6e5bffbdccb6903c813 onto `release/1.10.x` succeeded!

**Pull request merged**